packages/wrangler/src/deprecated/index.ts

```diff
@@ -47,25 +42,34 @@ export function generateOptions(yargs: Argv<CommonYargsOptions>) {
 			deprecated: true,
 		});
 }
+type GenerateArgs = YargsOptionsToInterface<typeof generateOptions>;
 
-export async function generateHandler({
+//TODO: move this out of deprecated since apparently it's not deprecated anymore
+export async function generateHandler(args: GenerateArgs) {
 	// somehow, `init` marks name as required but then also runs fine
 	// with the name omitted, and then substitutes it at runtime with ""
-	name = "",
-	template,
-	type,
-	site,
-	...args
-}: ArgumentsCamelCase<GenerateArgs>) {
 	// delegate to `wrangler init` if no template is specified
-	if (template === undefined) {
-		return initHandler({ name, ...args });
+	if (args.template === undefined) {
+		return initHandler({
+			name: args.name,
+			template: undefined,
+			site: undefined,
+			yes: undefined,
+			fromDash: undefined,
+			"from-dash": undefined,
+			v: undefined,
+			config: undefined,
+			env: undefined,
+			type: undefined,
+			_: args._,
+			$0: args.$0,
```

Comment:
I am so much not a fan of this 😂 I assume yargs breaks without it?

Reply:
Yeah - and at the moment we're passing it through `...args`. At least this way we know it's there